File: Q1b.py
sum=0
nofdist=0
#displaying the grid
for i in range(1,14):
    if (i>1): print('')
    for j in range(1,7):
        if (i%2!=0): print("  |  ",end='')
        else:  print("--o--",end='')
#calculating sum of all distances and no. of distances
for i in range(1,7):
    for j in range(1,7):
        for k in range(1,7):
            for l in range(1,7):
                sum+=abs(i-k)+abs(j-l)
                nofdist+=1
print("\nThe average distance between two points is: ",sum/nofdist)

File: Q1b.cpp
#include<iostream>
using namespace std;
int main(){
    float sum=0;
    float nofdist=0;
    //displaying the grid
    for (int i=1;i<=13;i++){
        if (i>1) cout<<endl;
        for(int j=1;j<=6;j++){
            if (i%2!=0) cout<<"  |  ";
            else cout<<"--o--";
        }
    }
    //calculating sum of all distances and no. of distances
    for (int i=1;i<=6;i++){
        for(int j=1;j<=6;j++){
            for(int k=1; k<=6;k++){
                for(int l=1;l<=6;l++){
                    sum+= abs(i-k) + abs(j-l);
                    nofdist+=1;
                }
            }

        }
    }
cout<<endl<<"The average distance between two points is: "<<sum/nofdist;
return 0;
}